services/motion/builtin/move_request.go

```diff
@@ -262,54 +267,41 @@ func (mr *moveRequest) obstaclesIntersectPlan(
 				return state.ExecuteResponse{}, err
 			}
 
-			// build representation of frame system's inputs
-			currentInputs, err := mr.kinematicBase.CurrentInputs(ctx)
-			if err != nil {
-				return state.ExecuteResponse{}, err
-			}
-			inputMap := referenceframe.StartPositions(mr.planRequest.FrameSystem)
-			inputMap[mr.kinematicBase.Name().ShortName()] = currentInputs
-
-			// get the current position of the base
-			currentPosition, err := mr.kinematicBase.CurrentPosition(ctx)
+			// get the execution state of the base
+			baseExecutionState, err := mr.kinematicBase.ExecutionState(ctx)
 			if err != nil {
 				return state.ExecuteResponse{}, err
 			}
 
-			// Note: the value of wayPointIndex is subject to change between when this function is first entered
-			// versus when CheckPlan is actually called.
-			// We load the wayPointIndex value to ensure that all information is up to date.
-
-			// get the pose difference between where the robot is versus where it ought to be.
-			errorState, err := mr.kinematicBase.ErrorState(ctx)
-			if err != nil {
-				return state.ExecuteResponse{}, err
-			}
+			// build representation of frame system's inputs
+			// TODO(pl): in the case where we have e.g. an arm (not moving) mounted on a base, we should be passing its current
+			// configuration rather than the zero inputs
+			inputMap := referenceframe.StartPositions(mr.planRequest.FrameSystem)
+			inputMap[mr.kinematicBase.Name().ShortName()] = baseExecutionState.CurrentInputs()[mr.kinematicBase.Name().ShortName()]
+			executionState := motionplan.NewExecutionState(
+				baseExecutionState.Plan(),
+				baseExecutionState.Index(),
+				inputMap,
+				baseExecutionState.CurrentPoses(),
+			)
 
-			mr.logger.CDebugf(ctx, "CheckPlan inputs: \n currentPosition: %v\n currentInputs: %v\n errorState: %v\n worldstate: %s",
-				spatialmath.PoseToProtobuf(currentPosition.Pose()),
-				currentInputs,
-				spatialmath.PoseToProtobuf(errorState),
+			mr.logger.CDebugf(ctx, "CheckPlan inputs: \n currentPosition: %v\n currentInputs: %v\n worldstate: %s",
```

Comment:
[opt]: preference for having this be within `CheckPlan`

Reply:
Disagree, logs should come from services not library functions in my opinion.

Reply:
sgtm.